microraiden/microraiden/test/test_http_client.py

```diff
@@ -536,3 +539,49 @@ def test_requests(
         response = microraiden.requests.get(url)
 
     assert response.text == 'success'
+
+
+def test_cooperative_close_denied(
+        default_http_client: DefaultHTTPClient,
+        api_endpoint_address: str,
+        token_address: str,
+        channel_manager_address: str,
+        receiver_address: str
+):
+    cooperative_close_denied_mock = mock.patch.object(
+        default_http_client,
+        'on_cooperative_close_denied',
+        wraps=default_http_client.on_cooperative_close_denied
+    ).start()
+
+    with requests_mock.mock() as server_mock:
+        headers = {
+            HTTPHeaders.TOKEN_ADDRESS: token_address,
+            HTTPHeaders.CONTRACT_ADDRESS: channel_manager_address,
+            HTTPHeaders.RECEIVER_ADDRESS: receiver_address,
+            HTTPHeaders.PRICE: '3'
+        }
+        headers = [headers.copy() for i in range(2)]
+
+        i = 0
+
+        i += 1
+        headers[i][HTTPHeaders.COST] = '3'
+
+        url = 'http://{}/something'.format(api_endpoint_address)
+        server_mock.get(url, [
+            {'status_code': 402, 'headers': headers[0]},
+            {'status_code': 200, 'headers': headers[1], 'text': 'success'},
+        ])
+        channel_url = re.compile(
+            'http://{}/api/1/channels/0x.{{40}}/\d+'.format(api_endpoint_address)
+        )
+        server_mock.delete(channel_url, [
```

Comment:
Just a note: atm we have two `DELETE` methods implemented. One for `/channels/<sender_addr>/<opening_block>` and the other for `/channels/<sender_addr>`

We should make them consistent - one returns text string, the other JSON.

Reply:
Which one does `/channels/<sender_addr>` default to if there are multiple channels open?

Reply:
it requires `{'open_block' : ..., 'signature': ... }` json as a request data